### DISCUSSION TITLE
Grammar. Complete and correct?

### DISCUSSION THREAD

**Discussion by B1Z0N** [takegue/tree-sitter-sql-bigquery#33] [Q&A] [answered]
I'd like to know if this grammar is the complete and correct version of the latest GoogleSQL(I mean BigQuery). And if the reader doesn't know, then I'll be glad to hear some thoughts on how I may check it the most efficiently. Because I'm doing that, but just looking at the spec and at the grammar actually takes a lot of time.

**Comment by takegue** [comment, marked as answer]
Hello this is the author. 
This project doesn't have perfect support for latest GoogleSQL version (including beta features), but it covers a lot of syntax.
If you want perfect AST parser, it is better to use ZetaSQL(https://github.com/google/zetasql). (Supported by Google)

You could get parsed AST results demo as examples, or find test cases in https://github.com/takegue/tree-sitter-sql-bigquery/tree/main/test/corpus. Given unparseable SQL, this outputs syntax tree including `ERROR` node.
If you find some errors, plz tell me via issues :)

Thank you.
